File: src/TwoDOF/Prog4/Forcedresponse/Forcedresponse.java
package TwoDOF.Prog4.Forcedresponse;

import java.applet.Applet;
import java.awt.BorderLayout;
import java.awt.Image;
import java.awt.Label;
import java.awt.Panel;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.util.Date;
import javax.swing.Timer;

public class Forcedresponse extends Applet implements ActionListener {
   static final double kMinK2divK1 = 0.2;
   static final double kMaxK2divK1 = 5.0;
   static final double kMinM2divM1 = 0.2;
   static final double kMaxM2divM1 = 5.0;
   static final String kRunString = "Run";
   CFramePanel mFramePanel;
   CFrameAnimation mAnimFrame;
   Timer mTimer = null;
   long mLastTime = 0;

   public void init() {
      this.setLayout(new BorderLayout());
      Panel buttonPanel = new Panel();
      this.add("North", buttonPanel);
      Label copyright = new Label("  This program is copyright, you need permission to use it. © 2004 B J Stone, bjs@mech.uwa.edu.au.");
      this.add("South", copyright);
      this.mFramePanel = new CFramePanel(this);
      this.add("Center", this.mFramePanel);
      this.mAnimFrame = new CFrameAnimation(this.mFramePanel, 0, 0, this.getSize().width, this.getSize().height);
      CFrameSmallControl K2divK1Control = new CFrameSmallControl(this.mFramePanel, this.mAnimFrame, 1, 450, 10, 0.2, 2.0, 5.0, " ");
      K2divK1Control.SetLabelOffset(20, 10);
      K2divK1Control.SetTextEditable(true);
      Image K2divK1Image = this.getImage(this.getCodeBase(), "K2divK1.gif");
      new CFramePicture(this.mFramePanel, 471, 25, 0, 0, K2divK1Image, false);
      CFrameSmallControl M2divM1Control = new CFrameSmallControl(this.mFramePanel, this.mAnimFrame, 2, 520, 10, 0.2, 0.5, 5.0, " ");
      M2divM1Control.SetLabelOffset(20, 10);
      M2divM1Control.SetTextEditable(true);
      Image M2divM1Image = this.getImage(this.getCodeBase(), "M2divM1.gif");
      new CFramePicture(this.mFramePanel, 541, 25, 0, 0, M2divM1Image, false);
      Image DiagramImage = this.getImage(this.getCodeBase(), "Diagram.gif");
      new CFramePicture(this.mFramePanel, 460, 60, 0, 0, DiagramImage, false);
      Image XaxisImage = this.getImage(this.getCodeBase(), "Xaxis.gif");
      new CFramePicture(this.mFramePanel, 96, 145, 0, 0, XaxisImage, false);
      new CFramePicture(this.mFramePanel, 96, 465, 0, 0, XaxisImage, false);
      Image Y1axesImage = this.getImage(this.getCodeBase(), "Y1axis.gif");
      new CFramePicture(this.mFramePanel, 20, 5, 0, 0, Y1axesImage, false);
      Image Y2axesImage = this.getImage(this.getCodeBase(), "Y2axis.gif");
      new CFramePicture(this.mFramePanel, 20, 325, 0, 0, Y2axesImage, false);
      Image LegendImage = this.getImage(this.getCodeBase(), "Legend.gif");
      new CFramePicture(this.mFramePanel, 280, 25, 0, 0, LegendImage, false);
      new CFramePicture(this.mFramePanel, 280, 345, 0, 0, LegendImage, false);
      this.validate();
   }

   public void start() {
      this.mTimer = new Timer(20, this);
      this.mTimer.start();
   }

   public void actionPerformed(ActionEvent e) {
      if (this.mTimer != null) {
         Date ddd = new Date();
         long thisTime = ddd.getTime();
         
         if (this.mLastTime == 0) {
            this.mLastTime = thisTime;
         }
         double realSeconds = (thisTime - this.mLastTime) / 1000.0;
         this.mLastTime = thisTime;
         
         //this.mAnimFrame.ControlMessage(0, realSeconds);
      }
   }
}

File: site/swingjs/j2s/TwoDOF/Prog4/Forcedresponse/Forcedresponse.js
(function(){var P$=Clazz.newPackage("TwoDOF.Prog4.Forcedresponse"),I$=[[0,'java.awt.BorderLayout','java.awt.Panel','java.awt.Label','TwoDOF.Prog4.Forcedresponse.CFramePanel','TwoDOF.Prog4.Forcedresponse.CFrameAnimation','TwoDOF.Prog4.Forcedresponse.CFrameSmallControl','TwoDOF.Prog4.Forcedresponse.CFramePicture','javax.swing.Timer']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "Forcedresponse", null, 'java.applet.Applet', 'java.awt.event.ActionListener');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
this.mTimer=null;
this.mLastTime=0;
},1);

C$.$fields$=[['J',['mLastTime'],'O',['mFramePanel','TwoDOF.Prog4.Forcedresponse.CFramePanel','mAnimFrame','TwoDOF.Prog4.Forcedresponse.CFrameAnimation','mTimer','javax.swing.Timer']]]

Clazz.newMeth(C$, ['init$','init'],  function () {
this.setLayout$java_awt_LayoutManager(Clazz.new_($I$(1,1)));
var buttonPanel=Clazz.new_($I$(2,1));
this.add$S$java_awt_Component("North", buttonPanel);
var copyright=Clazz.new_($I$(3,1).c$$S,["  This program is copyright, you need permission to use it. \u00a9 2004 B J Stone, bjs@mech.uwa.edu.au."]);
this.add$S$java_awt_Component("South", copyright);
this.mFramePanel=Clazz.new_($I$(4,1).c$$java_applet_Applet,[this]);
this.add$S$java_awt_Component("Center", this.mFramePanel);
this.mAnimFrame=Clazz.new_([this.mFramePanel, 0, 0, this.getSize$().width, this.getSize$().height],$I$(5,1).c$$TwoDOF_Prog4_Forcedresponse_CFramePanel$I$I$I$I);
var K2divK1Control=Clazz.new_($I$(6,1).c$$TwoDOF_Prog4_Forcedresponse_CFramePanel$TwoDOF_Prog4_Forcedresponse_CFrame$I$I$I$D$D$D$S,[this.mFramePanel, this.mAnimFrame, 1, 450, 10, 0.2, 2.0, 5.0, " "]);
K2divK1Control.SetLabelOffset$I$I(20, 10);
K2divK1Control.SetTextEditable$Z(true);
var K2divK1Image=this.getImage$java_net_URL$S(this.getCodeBase$(), "K2divK1.gif");
Clazz.new_($I$(7,1).c$$TwoDOF_Prog4_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 471, 25, 0, 0, K2divK1Image, false]);
var M2divM1Control=Clazz.new_($I$(6,1).c$$TwoDOF_Prog4_Forcedresponse_CFramePanel$TwoDOF_Prog4_Forcedresponse_CFrame$I$I$I$D$D$D$S,[this.mFramePanel, this.mAnimFrame, 2, 520, 10, 0.2, 0.5, 5.0, " "]);
M2divM1Control.SetLabelOffset$I$I(20, 10);
M2divM1Control.SetTextEditable$Z(true);
var M2divM1Image=this.getImage$java_net_URL$S(this.getCodeBase$(), "M2divM1.gif");
Clazz.new_($I$(7,1).c$$TwoDOF_Prog4_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 541, 25, 0, 0, M2divM1Image, false]);
var DiagramImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "Diagram.gif");
Clazz.new_($I$(7,1).c$$TwoDOF_Prog4_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 460, 60, 0, 0, DiagramImage, false]);
var XaxisImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "Xaxis.gif");
Clazz.new_($I$(7,1).c$$TwoDOF_Prog4_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 96, 145, 0, 0, XaxisImage, false]);
Clazz.new_($I$(7,1).c$$TwoDOF_Prog4_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 96, 465, 0, 0, XaxisImage, false]);
var Y1axesImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "Y1axis.gif");
Clazz.new_($I$(7,1).c$$TwoDOF_Prog4_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 20, 5, 0, 0, Y1axesImage, false]);
var Y2axesImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "Y2axis.gif");
Clazz.new_($I$(7,1).c$$TwoDOF_Prog4_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 20, 325, 0, 0, Y2axesImage, false]);
var LegendImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "Legend.gif");
Clazz.new_($I$(7,1).c$$TwoDOF_Prog4_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 280, 25, 0, 0, LegendImage, false]);
Clazz.new_($I$(7,1).c$$TwoDOF_Prog4_Forcedresponse_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 280, 345, 0, 0, LegendImage, false]);
this.validate$();
});

Clazz.newMeth(C$, ['start$','start'],  function () {
this.mTimer=Clazz.new_($I$(8,1).c$$I$java_awt_event_ActionListener,[20, this]);
this.mTimer.start$();
});

Clazz.newMeth(C$, ['actionPerformed$java_awt_event_ActionEvent','actionPerformed'],  function (e) {
if (this.mTimer != null ) {
var ddd=Clazz.new_(java.util.Date);
var thisTime=ddd.getTime$();
if (Long.$eq(this.mLastTime,0 )) {
this.mLastTime=thisTime;
}this.mLastTime=thisTime;
}});

Clazz.newMeth(C$);
})();
;Clazz.setTVer('5.0.1-v7');//Created 2025-10-14 19:43:52 Java2ScriptVisitor version 5.0.1-v7 net.sf.j2s.core.jar version 5.0.1-v7
